tracer/src/Datadog.Trace/AppSec/AttackerFingerprint/AttackerFingerprintHelper.cs

```diff
@@ -0,0 +1,58 @@
+// <copyright file="AttackerFingerprintHelper.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.Collections.Generic;
+using Datadog.Trace.AppSec.Coordinator;
+using Datadog.Trace.AppSec.Waf;
+using Datadog.Trace.Logging;
+
+#nullable enable
+
+namespace Datadog.Trace.AppSec.AttackerFingerprint;
+
+internal static class AttackerFingerprintHelper
+{
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(AttackerFingerprintHelper));
+    private static readonly Dictionary<string, object> _fingerprintRequest = new() { { AddressesConstants.WafContextProcessor, new Dictionary<string, object> { { "fingerprint", true } } } };
+
+    public static void AddSpanTags(Span span)
+    {
+        if (span.IsFinished || span.Type != SpanTypes.Web)
+        {
+            return;
+        }
+
+        var securityCoordinator = new SecurityCoordinator(Security.Instance, span);
+
+        // We need a context
+        if (!securityCoordinator.HasContext() || securityCoordinator.IsAdditiveContextDisposed())
+        {
+            return;
+        }
+
+        var result = securityCoordinator.RunWaf(_fingerprintRequest);
+        AddSpanTags(result?.FingerprintDerivatives, span);
+    }
+
+    private static void AddSpanTags(Dictionary<string, object?>? fingerPrintDerivatives, ISpan span)
+    {
+        if (fingerPrintDerivatives is not null)
+        {
+            foreach (var derivative in fingerPrintDerivatives)
+            {
+                var value = derivative.Value?.ToString();
+                if (!string.IsNullOrEmpty(value))
+                {
+                    span.SetTag(derivative.Key, value);
+                }
+                else
+                {
+                    // This should not happen
+                    Log.Warning("Fingerprint derivative {DerivativeKey} has no value", derivative.Key);
```

Comment:
Is there any concern that if we _do_ hit this for some weird reason, then we could end up writing a _lot_ of logs? If we have any concerns about it, we could _either_
1. convert this to debug
2. Add a flag to make sure we only right this once?

Reply:
I think that the second option it's better. If it happens, we should know but we don't want to flood the log. Thanks!